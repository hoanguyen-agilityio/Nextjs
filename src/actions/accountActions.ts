'use server';
import { signIn, signOut } from '@/configs';
import { ACCOUNT_URL, MESSAGE, ROUTERS } from '@/constants';
import { APIs } from '@/services';
import { Account } from '@/types';

const getAccount = async (): Promise<Account[]> => {
  try {
    if (!ACCOUNT_URL) {
      throw new Error(MESSAGE.ERROR_URL);
    }

    return await APIs.get(undefined, undefined, ACCOUNT_URL);
  } catch (error) {
    console.error(MESSAGE.ERROR_GET_ACCOUNT, error);
    return [];
  }
};

const handleSignIn = async (payload: Account) => {
  try {
    const result = await signIn('credentials', {
      ...payload,
      redirect: false,
    });

    if (result?.error) {
      return { error: result.error };
    }

    return { success: true };
  } catch (error) {
    return { error: MESSAGE.INVALID_ACCOUNT };
  }
};

const handleSignOut = async () => {
  await signOut({ redirectTo: `/${ROUTERS.LOGIN}` });
};

export { getAccount, handleSignIn, handleSignOut };
