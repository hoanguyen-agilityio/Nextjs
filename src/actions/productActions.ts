'use server';
import {
  EMPTY_TEXT,
  MESSAGE,
  OVERVIEW_URL,
  PRODUCT_URL,
  ROUTERS,
} from '@/constants';
import { APIs } from '@/services';
import { IProducts, OverviewDataItem } from '@/types';

const getDataProducts = async (
  search = EMPTY_TEXT,
): Promise<IProducts[] | null> => {
  try {
    if (!PRODUCT_URL) {
      throw new Error(MESSAGE.ERROR_URL);
    }

    const url = search.trim()
      ? `?search=${encodeURIComponent(search)}`
      : ROUTERS.EMPTY;

    return await APIs.get(url);
  } catch (error) {
    console.error(MESSAGE.ERROR_GET_DATA_PRODUCT, error);
    return null;
  }
};

const getDataOverview = async () => {
  const revalidate = 60;
  try {
    if (!OVERVIEW_URL) {
      throw new Error(MESSAGE.ERROR_URL);
    }
    const res = await APIs.get<OverviewDataItem[]>(
      undefined,
      revalidate,
      OVERVIEW_URL,
    );
    return res;
  } catch (error) {
    console.error(MESSAGE.ERROR_GET_DATA_OVERVIEW, error);
    return [];
  }
};

const handleAddProduct = async (products: IProducts) => {
  try {
    if (!PRODUCT_URL) {
      throw new Error(MESSAGE.ERROR_URL);
    }
    await APIs.post(ROUTERS.EMPTY, products);
  } catch (error) {
    const errorMessage = (error as Error).message;
    throw new Error(errorMessage);
  }
};

const handleEditProduct = async (id: string, products: IProducts) => {
  try {
    await APIs.put(`/${id}`, products);
  } catch (error) {
    const errorMessage = (error as Error).message;
    throw new Error(errorMessage);
  }
};

const handleDeleteProduct = async (id: string) => {
  try {
    const res = await APIs.delete(`/${id}`);
    return res;
  } catch (error) {
    const errorMessage = (error as Error).message;
    throw new Error(errorMessage);
  }
};

export {
  getDataProducts,
  getDataOverview,
  handleAddProduct,
  handleEditProduct,
  handleDeleteProduct,
};
