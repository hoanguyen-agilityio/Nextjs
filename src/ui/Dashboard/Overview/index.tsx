// Constants
import { dateDicker, productSelector } from '@/mocks';

// API
import { getDataOverview } from '@/actions';

// Components
import { RenderSelect } from '../Select';
import { OverviewStats } from '../OverviewStats';
import { CalendarIcon } from '@/icons';

const Overview = async () => {
  const dataOverview = (await getDataOverview()) || [];

  return (
    <div className="overflow-x-auto">
      <section className="w-full rounded-lg bg-white dark:bg-dark-900 min-w-[1200px]">
        <div className="flex pt-5 pl-5">
          <div className="pl-lg pr-5 py-3 rounded-lg flex justify-between  border-1 border-gray-50 w-s cursor-pointer dark:border-gray-700">
            <CalendarIcon width="20px" height="21px" fill="black" />
            <span className="font-semibold text-sm">
              12/12/2022 - 12/20/2022
            </span>
          </div>
          <RenderSelect
            defaultSelectedKey="weekly"
            items={dateDicker}
            className="w-[120px]"
            ariaLabel="date selector"
          />
          <RenderSelect
            defaultSelectedKey="allProducts"
            items={productSelector}
            className="w-[152px]"
            ariaLabel="product selector"
          />
        </div>
        <OverviewStats data={dataOverview} />
      </section>
    </div>
  );
};

export default Overview;
