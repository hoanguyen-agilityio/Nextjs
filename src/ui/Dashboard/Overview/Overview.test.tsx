import { render, screen, waitFor } from '@testing-library/react';
import Overview from '.';
import { getDataOverview } from '@/actions';

jest.mock('@/actions', () => ({
  getDataOverview: jest.fn(),
}));

describe('Overview Component', () => {
  test('renders Overview with mock data', async () => {
    const { container } = render(await Overview());
    expect(container).toMatchSnapshot();
  });

  test('renders Overview when no data is returned', async () => {
    // Arrange: Mock the API to return an empty array (no data)
    (getDataOverview as jest.Mock).mockResolvedValue([]);

    // Act: Render the component
    const { container } = render(await Overview());

    // Wait for the data to be fetched and the component to render
    await waitFor(() => expect(getDataOverview).toHaveBeenCalledTimes(1));

    // Assert: Check the snapshot
    expect(container).toMatchSnapshot();

    // Optionally, check if no data is displayed
    expect(screen.queryByText('Test value 1')).toBeNull();
    expect(screen.queryByText('Label 1')).toBeNull();
  });
});
