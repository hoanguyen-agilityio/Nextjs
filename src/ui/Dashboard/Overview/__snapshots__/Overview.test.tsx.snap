// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview Component renders Overview when no data is returned 1`] = `
<div>
  <div
    class="overflow-x-auto"
  >
    <section
      class="w-full rounded-lg bg-white dark:bg-dark-900 min-w-[1200px]"
    >
      <div
        class="flex pt-5 pl-5"
      >
        <div
          class="pl-lg pr-5 py-3 rounded-lg flex justify-between  border-1 border-gray-50 w-s cursor-pointer dark:border-gray-700"
        >
          <svg
            fill="none"
            height="21px"
            viewBox="0 0 21 23"
            width="20px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="fill-black-900 dark:fill-white"
              clip-rule="evenodd"
              d="M6.38791 22.2366C2.65292 22.2366 0.503906 20.0916 0.503906 16.3646V7.68764C0.503906 4.22363 2.35989 2.12663 5.62091 1.84663V1.04663C5.63577 0.633514 5.97501 0.30629 6.3884 0.30629C6.80179 0.30629 7.14102 0.633514 7.15588 1.04663L7.15591 1.81264H13.8069V1.04663C13.8078 0.623657 14.1509 0.28125 14.5739 0.28125C14.9969 0.28125 15.34 0.623657 15.3409 1.04663V1.84663C18.6009 2.12564 20.4579 4.22264 20.4579 7.68764V16.3686C20.4579 20.0966 18.3089 22.2406 14.5739 22.2406L6.38791 22.2366ZM2.03891 16.3646C2.03891 19.2856 3.46091 20.7056 6.38791 20.7056H14.5729C17.4999 20.7056 18.9229 19.2866 18.9229 16.3646V9.05264H2.03891V16.3646ZM18.9209 7.52065C18.8729 4.97264 17.6989 3.61165 15.3409 3.37964V4.10963C15.34 4.53261 14.9969 4.87502 14.5739 4.87502C14.1509 4.87502 13.8078 4.53261 13.8069 4.10963V3.34363H7.15591V4.10963C7.14102 4.52275 6.80179 4.84998 6.3884 4.84998C5.97501 4.84998 5.63577 4.52275 5.62091 4.10963V3.37964C3.26291 3.61163 2.0889 4.97264 2.04092 7.52065H18.9209ZM13.3349 17.0906C13.24 16.9949 13.1667 16.88 13.1199 16.7536C13.0107 16.5065 13.0107 16.2248 13.1199 15.9776C13.1669 15.8513 13.2402 15.7365 13.3349 15.6406C13.5772 15.4018 13.9223 15.2988 14.2559 15.3656C14.3235 15.3749 14.3891 15.3955 14.4499 15.4266C14.5152 15.4494 14.5774 15.4803 14.6349 15.5186C14.6883 15.5562 14.7393 15.597 14.7879 15.6406C15.173 16.0444 15.1652 16.6818 14.7704 17.0761C14.3756 17.4703 13.7382 17.4772 13.3349 17.0916V17.0906ZM9.75391 17.0906C9.56595 16.8955 9.4588 16.6365 9.45392 16.3656C9.45538 16.2322 9.48325 16.1003 9.53589 15.9776C9.58258 15.8512 9.65591 15.7363 9.75092 15.6406C10.1614 15.2623 10.7935 15.2623 11.2039 15.6406C11.2986 15.7365 11.3719 15.8514 11.4189 15.9776C11.4716 16.1003 11.4994 16.2322 11.5009 16.3656C11.4982 16.7789 11.2482 17.1504 10.8664 17.3086C10.4846 17.4668 10.0451 17.3809 9.75092 17.0906H9.75391ZM6.50992 17.3046C6.38586 17.2531 6.27155 17.1807 6.17191 17.0906C5.98395 16.8955 5.8768 16.6365 5.87192 16.3656C5.87387 16.2322 5.90173 16.1004 5.95395 15.9776C6.00098 15.8513 6.07422 15.7365 6.16891 15.6406C6.49588 15.3072 7.00659 15.2363 7.41199 15.4681C7.81738 15.6999 8.01535 16.176 7.89383 16.6269C7.77231 17.0778 7.36191 17.3899 6.89493 17.3866C6.76242 17.3842 6.63162 17.356 6.50992 17.3036V17.3046ZM6.50992 13.7306C6.12057 13.5834 5.86621 13.2068 5.87491 12.7906C5.87677 12.6572 5.9046 12.5254 5.95691 12.4026C6.00903 12.279 6.08173 12.165 6.17194 12.0656C6.27017 11.9737 6.38483 11.9011 6.50992 11.8516C6.89163 11.6973 7.32837 11.7811 7.62592 12.0656C7.8136 12.2609 7.92072 12.5198 7.9259 12.7906C7.92349 12.8577 7.91681 12.9245 7.90594 12.9906C7.89426 13.0547 7.87338 13.1166 7.84387 13.1747C7.82205 13.24 7.79111 13.302 7.75189 13.3586C7.71451 13.4122 7.67374 13.4633 7.62988 13.5116C7.43411 13.6998 7.17441 13.807 6.90289 13.8116C6.76776 13.8106 6.63419 13.7827 6.50992 13.7296V13.7306ZM13.6729 13.7306C13.5479 13.6811 13.4335 13.6082 13.3359 13.5156L13.2129 13.3626C13.1744 13.3055 13.1435 13.2437 13.1209 13.1786C13.0913 13.1208 13.0707 13.0587 13.0599 12.9946C13.0486 12.9285 13.0417 12.8617 13.0389 12.7946C13.0436 12.5237 13.1508 12.2646 13.3389 12.0696C13.4363 11.9771 13.5508 11.9044 13.6759 11.8556C14.0567 11.695 14.4967 11.7794 14.7909 12.0696C14.979 12.2646 15.0862 12.5237 15.0909 12.7946C15.0886 12.8617 15.0816 12.9285 15.0699 12.9946C15.0591 13.0587 15.0385 13.1208 15.0089 13.1786C14.9863 13.2437 14.9554 13.3055 14.9169 13.3626C14.8794 13.4164 14.8383 13.4675 14.7939 13.5156C14.5983 13.7035 14.339 13.8106 14.0679 13.8156C13.9318 13.8134 13.7976 13.7841 13.6729 13.7296V13.7306ZM10.0919 13.7306C9.70258 13.5834 9.44821 13.2068 9.45691 12.7906C9.45837 12.6568 9.48627 12.5246 9.53891 12.4016C9.59061 12.2777 9.66333 12.1637 9.75391 12.0646C9.85162 11.972 9.96643 11.8993 10.0919 11.8506C10.4728 11.6908 10.9123 11.7752 11.2069 12.0646C11.395 12.2596 11.5022 12.5187 11.5069 12.7896C11.5042 12.8567 11.4972 12.9235 11.4859 12.9896C11.4751 13.0537 11.4545 13.1158 11.4249 13.1736C11.4023 13.2389 11.3711 13.3008 11.3319 13.3576C11.2945 13.4112 11.2538 13.4623 11.2099 13.5106C11.0142 13.6983 10.755 13.8054 10.4839 13.8106C10.3492 13.8095 10.216 13.782 10.0919 13.7296V13.7306Z"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="text-sm"
          >
            12/12/2022 - 12/20/2022
          </span>
        </div>
        <div
          class="group flex flex-col transition-background motion-reduce:transition-none !duration-150 group relative justify-end ml-5 w-[120px]"
          data-filled="true"
          data-has-value="true"
          data-slot="base"
        >
          <div
            aria-hidden="true"
            data-a11y-ignore="aria-hidden-focus"
            data-testid="hidden-select-container"
            style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
          >
            <input
              style="font-size: 16px;"
              tabindex="-1"
              type="text"
              value="weekly"
            />
            <label>
              <select
                size="4"
                tabindex="-1"
                title=""
              >
                <option />
                <option
                  value="day"
                >
                  Day
                </option>
                <option
                  value="weekly"
                >
                  Weekly
                </option>
                <option
                  value="month"
                >
                  Month
                </option>
                <option
                  value="year"
                >
                  Year
                </option>
              </select>
            </label>
          </div>
          <div
            class="w-full flex flex-col"
            data-slot="mainWrapper"
          >
            <button
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-label="date selector"
              aria-labelledby="react-aria-:rn: react-aria-:ri:"
              class="relative px-3 gap-3 w-full inline-flex flex-row items-center shadow-sm tap-highlight-transparent group-data-[focus=true]:bg-default-100 min-h-10 rounded-medium data-[hover=true]:bg-default-50 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 bg-white border-1 border-gray-50 dark:bg-dark-900 dark:border-gray-700 py-3.5 h-auto"
              data-slot="trigger"
              id="react-aria-:ri:"
              type="button"
            >
              <div
                class="inline-flex h-full w-[calc(100%_-_theme(spacing.6))] min-h-4 items-center gap-1.5 box-border"
                data-slot="innerWrapper"
              >
                <span
                  class="w-full text-left rtl:text-right text-small truncate group-data-[has-value=true]:text-default-foreground font-bold text-black"
                  data-slot="value"
                  id="react-aria-:rn:"
                >
                  Weekly
                </span>
              </div>
              <svg
                aria-hidden="true"
                class="absolute end-3 w-4 h-4 transition-transform duration-150 ease motion-reduce:transition-none data-[open=true]:rotate-180"
                data-slot="selectorIcon"
                fill="none"
                focusable="false"
                height="1em"
                role="presentation"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="1em"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="group flex flex-col transition-background motion-reduce:transition-none !duration-150 group relative justify-end ml-5 w-[152px]"
          data-filled="true"
          data-has-value="true"
          data-slot="base"
        >
          <div
            aria-hidden="true"
            data-a11y-ignore="aria-hidden-focus"
            data-testid="hidden-select-container"
            style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
          >
            <input
              style="font-size: 16px;"
              tabindex="-1"
              type="text"
              value="allProducts"
            />
            <label>
              <select
                size="2"
                tabindex="-1"
                title=""
              >
                <option />
                <option
                  value="allProducts"
                >
                  All Products
                </option>
                <option
                  value="products"
                >
                  Products
                </option>
              </select>
            </label>
          </div>
          <div
            class="w-full flex flex-col"
            data-slot="mainWrapper"
          >
            <button
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-label="product selector"
              aria-labelledby="react-aria-:rv: react-aria-:rq:"
              class="relative px-3 gap-3 w-full inline-flex flex-row items-center shadow-sm tap-highlight-transparent group-data-[focus=true]:bg-default-100 min-h-10 rounded-medium data-[hover=true]:bg-default-50 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 bg-white border-1 border-gray-50 dark:bg-dark-900 dark:border-gray-700 py-3.5 h-auto"
              data-slot="trigger"
              id="react-aria-:rq:"
              type="button"
            >
              <div
                class="inline-flex h-full w-[calc(100%_-_theme(spacing.6))] min-h-4 items-center gap-1.5 box-border"
                data-slot="innerWrapper"
              >
                <span
                  class="w-full text-left rtl:text-right text-small truncate group-data-[has-value=true]:text-default-foreground font-bold text-black"
                  data-slot="value"
                  id="react-aria-:rv:"
                >
                  All Products
                </span>
              </div>
              <svg
                aria-hidden="true"
                class="absolute end-3 w-4 h-4 transition-transform duration-150 ease motion-reduce:transition-none data-[open=true]:rotate-180"
                data-slot="selectorIcon"
                fill="none"
                focusable="false"
                height="1em"
                role="presentation"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="1em"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="flex justify-between w-full pt-11 pb-5xl pl-20 pr-8xl bg-white text-black dark:bg-dark-900"
      >
        <p
          class="text-gray-500 w-full text-center"
        >
          No data available
        </p>
      </div>
    </section>
  </div>
</div>
`;

exports[`Overview Component renders Overview with mock data 1`] = `
<div>
  <div
    class="overflow-x-auto"
  >
    <section
      class="w-full rounded-lg bg-white dark:bg-dark-900 min-w-[1200px]"
    >
      <div
        class="flex pt-5 pl-5"
      >
        <div
          class="pl-lg pr-5 py-3 rounded-lg flex justify-between  border-1 border-gray-50 w-s cursor-pointer dark:border-gray-700"
        >
          <svg
            fill="none"
            height="21px"
            viewBox="0 0 21 23"
            width="20px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="fill-black-900 dark:fill-white"
              clip-rule="evenodd"
              d="M6.38791 22.2366C2.65292 22.2366 0.503906 20.0916 0.503906 16.3646V7.68764C0.503906 4.22363 2.35989 2.12663 5.62091 1.84663V1.04663C5.63577 0.633514 5.97501 0.30629 6.3884 0.30629C6.80179 0.30629 7.14102 0.633514 7.15588 1.04663L7.15591 1.81264H13.8069V1.04663C13.8078 0.623657 14.1509 0.28125 14.5739 0.28125C14.9969 0.28125 15.34 0.623657 15.3409 1.04663V1.84663C18.6009 2.12564 20.4579 4.22264 20.4579 7.68764V16.3686C20.4579 20.0966 18.3089 22.2406 14.5739 22.2406L6.38791 22.2366ZM2.03891 16.3646C2.03891 19.2856 3.46091 20.7056 6.38791 20.7056H14.5729C17.4999 20.7056 18.9229 19.2866 18.9229 16.3646V9.05264H2.03891V16.3646ZM18.9209 7.52065C18.8729 4.97264 17.6989 3.61165 15.3409 3.37964V4.10963C15.34 4.53261 14.9969 4.87502 14.5739 4.87502C14.1509 4.87502 13.8078 4.53261 13.8069 4.10963V3.34363H7.15591V4.10963C7.14102 4.52275 6.80179 4.84998 6.3884 4.84998C5.97501 4.84998 5.63577 4.52275 5.62091 4.10963V3.37964C3.26291 3.61163 2.0889 4.97264 2.04092 7.52065H18.9209ZM13.3349 17.0906C13.24 16.9949 13.1667 16.88 13.1199 16.7536C13.0107 16.5065 13.0107 16.2248 13.1199 15.9776C13.1669 15.8513 13.2402 15.7365 13.3349 15.6406C13.5772 15.4018 13.9223 15.2988 14.2559 15.3656C14.3235 15.3749 14.3891 15.3955 14.4499 15.4266C14.5152 15.4494 14.5774 15.4803 14.6349 15.5186C14.6883 15.5562 14.7393 15.597 14.7879 15.6406C15.173 16.0444 15.1652 16.6818 14.7704 17.0761C14.3756 17.4703 13.7382 17.4772 13.3349 17.0916V17.0906ZM9.75391 17.0906C9.56595 16.8955 9.4588 16.6365 9.45392 16.3656C9.45538 16.2322 9.48325 16.1003 9.53589 15.9776C9.58258 15.8512 9.65591 15.7363 9.75092 15.6406C10.1614 15.2623 10.7935 15.2623 11.2039 15.6406C11.2986 15.7365 11.3719 15.8514 11.4189 15.9776C11.4716 16.1003 11.4994 16.2322 11.5009 16.3656C11.4982 16.7789 11.2482 17.1504 10.8664 17.3086C10.4846 17.4668 10.0451 17.3809 9.75092 17.0906H9.75391ZM6.50992 17.3046C6.38586 17.2531 6.27155 17.1807 6.17191 17.0906C5.98395 16.8955 5.8768 16.6365 5.87192 16.3656C5.87387 16.2322 5.90173 16.1004 5.95395 15.9776C6.00098 15.8513 6.07422 15.7365 6.16891 15.6406C6.49588 15.3072 7.00659 15.2363 7.41199 15.4681C7.81738 15.6999 8.01535 16.176 7.89383 16.6269C7.77231 17.0778 7.36191 17.3899 6.89493 17.3866C6.76242 17.3842 6.63162 17.356 6.50992 17.3036V17.3046ZM6.50992 13.7306C6.12057 13.5834 5.86621 13.2068 5.87491 12.7906C5.87677 12.6572 5.9046 12.5254 5.95691 12.4026C6.00903 12.279 6.08173 12.165 6.17194 12.0656C6.27017 11.9737 6.38483 11.9011 6.50992 11.8516C6.89163 11.6973 7.32837 11.7811 7.62592 12.0656C7.8136 12.2609 7.92072 12.5198 7.9259 12.7906C7.92349 12.8577 7.91681 12.9245 7.90594 12.9906C7.89426 13.0547 7.87338 13.1166 7.84387 13.1747C7.82205 13.24 7.79111 13.302 7.75189 13.3586C7.71451 13.4122 7.67374 13.4633 7.62988 13.5116C7.43411 13.6998 7.17441 13.807 6.90289 13.8116C6.76776 13.8106 6.63419 13.7827 6.50992 13.7296V13.7306ZM13.6729 13.7306C13.5479 13.6811 13.4335 13.6082 13.3359 13.5156L13.2129 13.3626C13.1744 13.3055 13.1435 13.2437 13.1209 13.1786C13.0913 13.1208 13.0707 13.0587 13.0599 12.9946C13.0486 12.9285 13.0417 12.8617 13.0389 12.7946C13.0436 12.5237 13.1508 12.2646 13.3389 12.0696C13.4363 11.9771 13.5508 11.9044 13.6759 11.8556C14.0567 11.695 14.4967 11.7794 14.7909 12.0696C14.979 12.2646 15.0862 12.5237 15.0909 12.7946C15.0886 12.8617 15.0816 12.9285 15.0699 12.9946C15.0591 13.0587 15.0385 13.1208 15.0089 13.1786C14.9863 13.2437 14.9554 13.3055 14.9169 13.3626C14.8794 13.4164 14.8383 13.4675 14.7939 13.5156C14.5983 13.7035 14.339 13.8106 14.0679 13.8156C13.9318 13.8134 13.7976 13.7841 13.6729 13.7296V13.7306ZM10.0919 13.7306C9.70258 13.5834 9.44821 13.2068 9.45691 12.7906C9.45837 12.6568 9.48627 12.5246 9.53891 12.4016C9.59061 12.2777 9.66333 12.1637 9.75391 12.0646C9.85162 11.972 9.96643 11.8993 10.0919 11.8506C10.4728 11.6908 10.9123 11.7752 11.2069 12.0646C11.395 12.2596 11.5022 12.5187 11.5069 12.7896C11.5042 12.8567 11.4972 12.9235 11.4859 12.9896C11.4751 13.0537 11.4545 13.1158 11.4249 13.1736C11.4023 13.2389 11.3711 13.3008 11.3319 13.3576C11.2945 13.4112 11.2538 13.4623 11.2099 13.5106C11.0142 13.6983 10.755 13.8054 10.4839 13.8106C10.3492 13.8095 10.216 13.782 10.0919 13.7296V13.7306Z"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="text-sm"
          >
            12/12/2022 - 12/20/2022
          </span>
        </div>
        <div
          class="group flex flex-col transition-background motion-reduce:transition-none !duration-150 group relative justify-end ml-5 w-[120px]"
          data-filled="true"
          data-has-value="true"
          data-slot="base"
        >
          <div
            aria-hidden="true"
            data-a11y-ignore="aria-hidden-focus"
            data-testid="hidden-select-container"
            style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
          >
            <input
              style="font-size: 16px;"
              tabindex="-1"
              type="text"
              value="weekly"
            />
            <label>
              <select
                size="4"
                tabindex="-1"
                title=""
              >
                <option />
                <option
                  value="day"
                >
                  Day
                </option>
                <option
                  value="weekly"
                >
                  Weekly
                </option>
                <option
                  value="month"
                >
                  Month
                </option>
                <option
                  value="year"
                >
                  Year
                </option>
              </select>
            </label>
          </div>
          <div
            class="w-full flex flex-col"
            data-slot="mainWrapper"
          >
            <button
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-label="date selector"
              aria-labelledby="react-aria-:r7: react-aria-:r2:"
              class="relative px-3 gap-3 w-full inline-flex flex-row items-center shadow-sm tap-highlight-transparent group-data-[focus=true]:bg-default-100 min-h-10 rounded-medium data-[hover=true]:bg-default-50 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 bg-white border-1 border-gray-50 dark:bg-dark-900 dark:border-gray-700 py-3.5 h-auto"
              data-slot="trigger"
              id="react-aria-:r2:"
              type="button"
            >
              <div
                class="inline-flex h-full w-[calc(100%_-_theme(spacing.6))] min-h-4 items-center gap-1.5 box-border"
                data-slot="innerWrapper"
              >
                <span
                  class="w-full text-left rtl:text-right text-small truncate group-data-[has-value=true]:text-default-foreground font-bold text-black"
                  data-slot="value"
                  id="react-aria-:r7:"
                >
                  Weekly
                </span>
              </div>
              <svg
                aria-hidden="true"
                class="absolute end-3 w-4 h-4 transition-transform duration-150 ease motion-reduce:transition-none data-[open=true]:rotate-180"
                data-slot="selectorIcon"
                fill="none"
                focusable="false"
                height="1em"
                role="presentation"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="1em"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="group flex flex-col transition-background motion-reduce:transition-none !duration-150 group relative justify-end ml-5 w-[152px]"
          data-filled="true"
          data-has-value="true"
          data-slot="base"
        >
          <div
            aria-hidden="true"
            data-a11y-ignore="aria-hidden-focus"
            data-testid="hidden-select-container"
            style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
          >
            <input
              style="font-size: 16px;"
              tabindex="-1"
              type="text"
              value="allProducts"
            />
            <label>
              <select
                size="2"
                tabindex="-1"
                title=""
              >
                <option />
                <option
                  value="allProducts"
                >
                  All Products
                </option>
                <option
                  value="products"
                >
                  Products
                </option>
              </select>
            </label>
          </div>
          <div
            class="w-full flex flex-col"
            data-slot="mainWrapper"
          >
            <button
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-label="product selector"
              aria-labelledby="react-aria-:rf: react-aria-:ra:"
              class="relative px-3 gap-3 w-full inline-flex flex-row items-center shadow-sm tap-highlight-transparent group-data-[focus=true]:bg-default-100 min-h-10 rounded-medium data-[hover=true]:bg-default-50 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 bg-white border-1 border-gray-50 dark:bg-dark-900 dark:border-gray-700 py-3.5 h-auto"
              data-slot="trigger"
              id="react-aria-:ra:"
              type="button"
            >
              <div
                class="inline-flex h-full w-[calc(100%_-_theme(spacing.6))] min-h-4 items-center gap-1.5 box-border"
                data-slot="innerWrapper"
              >
                <span
                  class="w-full text-left rtl:text-right text-small truncate group-data-[has-value=true]:text-default-foreground font-bold text-black"
                  data-slot="value"
                  id="react-aria-:rf:"
                >
                  All Products
                </span>
              </div>
              <svg
                aria-hidden="true"
                class="absolute end-3 w-4 h-4 transition-transform duration-150 ease motion-reduce:transition-none data-[open=true]:rotate-180"
                data-slot="selectorIcon"
                fill="none"
                focusable="false"
                height="1em"
                role="presentation"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="1em"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="flex justify-between w-full pt-11 pb-5xl pl-20 pr-8xl bg-white text-black dark:bg-dark-900"
      >
        <p
          class="text-gray-500 w-full text-center"
        >
          No data available
        </p>
      </div>
    </section>
  </div>
</div>
`;
